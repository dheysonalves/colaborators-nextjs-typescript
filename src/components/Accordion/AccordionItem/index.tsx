import React, { useState } from 'react'
import {
	ContainerBorder,
	UserAvatar,
	CustomFlexContainer,
	Title,
	UserTitleName,
	Description,
	ButtoHeaderContainer,
	GridContainer,
} from './styles'
import { Container } from '../../utils/Flex/flex.styles'
import { MdKeyboardArrowDown, MdKeyboardArrowUp } from 'react-icons/md'
import StatusPin from '../../StatusPin'
import {
	AgentStatusTypes,
	DepartamentRolesTypes,
	NameRolesTypes,
} from '../../../api/@types/general.types'

export interface IAccordionItemProps {
	data: {
		agent_id: number
		name: string
		image: string
		department: DepartamentRolesTypes
		branch: string
		role: NameRolesTypes
		status: AgentStatusTypes
	}
}

const AccordionItem = ({ data }: IAccordionItemProps) => {
	const [toogle, setToogle] = useState(false)

	return (
		<ContainerBorder>
			<Container column>
				<Title tag="h1">Nome Completo</Title>
				<ButtoHeaderContainer
					onClick={() => setToogle(!toogle)}
					aria-label="button-toogle"
				>
					<CustomFlexContainer>
						<UserAvatar src={data.image ?? '/img/brand.svg'} />
						<UserTitleName tag="h1">{data.name}</UserTitleName>
					</CustomFlexContainer>
					{toogle ? (
						<MdKeyboardArrowDown size={24} color="#587169" />
					) : (
						<MdKeyboardArrowUp size={24} color="#587169" />
					)}
				</ButtoHeaderContainer>
				{toogle && (
					<GridContainer>
						<div>
							<Title tag="h1">Departamento</Title>
							<Description tag="p">{data.department}</Description>
						</div>
						<div>
							<Title tag="h1">Cargo</Title>
							<Description tag="p">{data.role}</Description>
						</div>
						<div>
							<Title tag="h1">Unidade</Title>
							<Description tag="p">{data.branch}</Description>
						</div>
						<div>
							<Title tag="h1">Status</Title>
							<StatusPin status={data.status} />
						</div>
					</GridContainer>
				)}
			</Container>
		</ContainerBorder>
	)
}

export default AccordionItem
