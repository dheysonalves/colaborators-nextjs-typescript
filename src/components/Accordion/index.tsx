import React from 'react'
import * as S from './styles'

const Accordion = () => (
	<S.Wrapper>
		<h1>Accordion</h1>
	</S.Wrapper>
)

export default Accordion
