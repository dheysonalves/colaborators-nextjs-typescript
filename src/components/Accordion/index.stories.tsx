import { ComponentMeta, ComponentStory } from '@storybook/react'
import Accordion from '.'

export default {
	title: 'Composites/Accordion',
	component: Accordion,
} as ComponentMeta<typeof Accordion>

export const Default: ComponentStory<typeof Accordion> = () => <Accordion />
