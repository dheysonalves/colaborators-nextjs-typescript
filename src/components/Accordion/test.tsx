import { render, screen } from '@testing-library/react'

import Accordion from '.'

describe('<Accordion />', () => {
	it('should render the heading', () => {
		const { container } = render(<Accordion />)

		expect(
			screen.getByRole('heading', { name: /Accordion/i }),
		).toBeInTheDocument()

		expect(container.firstChild).toMatchSnapshot()
	})
})
