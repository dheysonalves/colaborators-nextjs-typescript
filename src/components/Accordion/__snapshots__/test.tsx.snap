// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion /> should render the heading 1`] = `
<main
  class="css-0"
>
  <h1>
    Accordion
  </h1>
</main>
`
